doc/specs/#885 - winrt Terminal Settings.md

```diff
@@ -0,0 +1,211 @@
+---
+author: Carlos Zamora @carlos-zamora
+created on: 2020-07-10
+last updated: 2020-07-10
+issue id: [#885](https://github.com/microsoft/terminal/issues/885)
+---
+
+# Spec Title
+
+## Abstract
+
+This spec proposes a major refactor and repurposing of the TerminalSettings project. TerminalSettings would
+ be responsible for exposing, serializing, and deserializing settings as WinRT objects for Windows Terminal. In
+ doing so, Terminal's settings model is accessible as WinRT objects to existing components like TerminalApp,
+ TerminalControl, and TerminalCore. Additionally, Terminal Settings can be used by the Settings UI or
+ Shell Extensions to modify or reference Terminal's settings respectively.
+
+## Inspiration
+
+The main driver for this change is the Settings UI. The Settings UI will need to read and modify Terminal Settings.
+ At the time of writing this spec, the Terminal Settings are serialized as objects in the TerminalApp project.
+ To access these objects, the Settings UI would need to be a part of TerminalApp too, making it more bloated.
+
+## Solution Design
+
+### Terminal Settings Model: Objects and Projections
+
+The objects/interfaces introduced in this section will be exposed as WinRT objects/interfaces
+ within the `Microsoft.Terminal.Settings.Model` namespace. This allows them to be interacted with across

```

Comment:
At that point, I'd prefer to get rid of the post-fix "Settings" so that it looks like this:
- `Microsoft.Terminal.Settings.App`
- `Microsoft.Terminal.Settings.IControl`
- `Microsoft.Terminal.Settings.ICore`

@DHowett has been the main proponent for appending "Model" to the namespace. I don't recall why though but I'm fine either way.

Reply:
Going with `Microsoft.Terminal.Settings.Model` because we're working with an object model.

Additionally, when the Settings UI comes in, that'll be thrown into the `Microsoft.Terminal.Settings.Editor` namespace. (idk why this feels _oddly satisfying_ to me)